### PR TITLE
Refactor tradeparams, move events to relevant components

### PR DESCRIPTION
**Pull request by DanSallau**
Hi,

Attached contained changes that should refactore our tradeparams as mentioned in the card here https://trello.com/c/wcjZI13L/656-refactor-onchange-events-in-tradeparams-moved-to-each-param-component

### CODE REVIEW

**Comment by borisyankov** [comment]
@qingweibinary Review, please?

File: src/barrier-picker/BarrierCard.js [hunk 1]
```diff
@@ -16,11 +17,10 @@ export default class BarrierCard extends PureComponent {
         barrierInfo: PropTypes.object,
         barrierType: PropTypes.oneOf(['relative', 'absolute']),
         isIntraDay: PropTypes.bool,
-        onBarrier1Change: PropTypes.func,
-        onBarrier2Change: PropTypes.func,
-        onError: PropTypes.func,
         pipSize: PropTypes.number,
         spot: PropTypes.number,
+        actions: PropTypes.object,
```
Comment (qingweibinary) [inline src/barrier-picker/BarrierCard.js]: Now we do not pass actions as props, instead import them, refer to how we get actions in `TradeParams`

File: src/duration-picker/DurationCard.js [hunk 2]
```diff
@@ -11,9 +12,11 @@ export default class DurationCard extends PureComponent {
         durationUnit: PropTypes.string,
         forwardStartingDuration: PropTypes.object,       // treated as special case
         options: PropTypes.array,
-        onUnitChange: PropTypes.func,
-        onDurationChange: PropTypes.func,
-        onError: PropTypes.func,
+        index: PropTypes.number,
+        actions: PropTypes.object,
```
Comment (qingweibinary) [inline src/duration-picker/DurationCard.js]: Same, do not pass actions

File: src/payout-picker/StakeCard.js [hunk 3]
```diff
@@ -1,19 +1,43 @@
...
+            return;
+        }
+        actions.updateTradeError(index, 'stakeError', undefined);
+        const updatedAmount = changeAmount(inputValue);
+        onUpdateTradeParams(updatedAmount);
+    }, isMobile ? 300 : 150, { leading: true, trailing: true })
+
+    onAmountChange = e => {
+        const { actions } = this.props;
+        actions.updateTradeUIState(this.props.index, 'disabled', true);
```
Comment (qingweibinary) [inline src/payout-picker/StakeCard.js]: index can be extract in line above

**Comment by qingweibinary** [comment]
LGTM 👍

**Pull request merged**